### PR TITLE
Update kafka generator to use cookie from mappings

### PR DESCRIPTION
**Pull request by Dmitriy-Ivanov-04**

### CODE REVIEW

File: generators/src/main/java/kafka/pageclickgenerator/PageClickGenerator.java [hunk 1]
```diff
@@ -26,12 +26,15 @@ public PageClickGenerator(String urlsPath) throws FileNotFoundException {
...
+
+        // Read cookies from CookieEmailMappings file
+        cookies = GeneratorHelpers.readColumnFromCsvFile(cookieEmailMappingsPath, 0);
     }
 
     public PageClick getNext() {
         int rand = random.nextInt(length);
         PageClick pageClick = new PageClick();
-        pageClick.setCookie(UUID.randomUUID().toString());
+        pageClick.setCookie(getRandomCookieFromMappings());
```
Comment (S4RPYK3IVl5) [inline generators/src/main/java/kafka/pageclickgenerator/PageClickGenerator.java:37]: Why did you extracted this single line code at the different function?

File: generators/src/main/java/kafka/pageclickgenerator/PageClickGenerator.java [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
-    private Random random;
-
-    public PageClickGenerator(String urlsPath) throws FileNotFoundException {
-        urls = new ArrayList<>();
-        refererUrls = new ArrayList<>();
-        random = new Random();
+    private Random random = new Random();
 
+    public PageClickGenerator(String urlsPath, String cookieEmailMappingsPath) throws IOException, CsvValidationException {
+        // Read URls from the file so that we can randomly use them later
```
Comment (S4RPYK3IVl5) [inline generators/src/main/java/kafka/pageclickgenerator/PageClickGenerator.java:20]: Remove all commentaries, code have to be fully understandable by itself

File: generators/src/main/java/kafka/pageclickgenerator/PageClickGenerator.java [hunk 3]
```diff
@@ -26,12 +26,15 @@ public PageClickGenerator(String urlsPath) throws FileNotFoundException {
             refererUrls.add(split[1]);
             length += 2;
         }
+
+        // Read cookies from CookieEmailMappings file
```
Comment (S4RPYK3IVl5) [inline generators/src/main/java/kafka/pageclickgenerator/PageClickGenerator.java:30]: Comment

File: generators/src/main/java/kafka/pageclickgenerator/Main.java [hunk 4]
```diff
@@ -9,6 +9,7 @@ public static void main(String[] args) throws Exception {
         String bootstrapServer = KafkaUtils.BOOTSTRAP_SERVERS;
         String topic = KafkaUtils.TOPIC;
         String urlsPath = "generators/src/main/resources/urls.csv";
+        String cookieEmailMappingsPath = "generators/src/main/resources/userCookieEmailMappings.csv";
         if (args.length >= 2) {
             bootstrapServer = args[0];
             topic = args[1];
```
Comment (Levshunovma) [inline generators/src/main/java/kafka/pageclickgenerator/Main.java:15]: Should we add possibility to get cookieEmailMappingsPath from program arguments like we do with bootstrapServer, topic and urlsPath? Or will we use this class only from IDE?
P.S. This comment refers to lines 12-19, but Github doesn't allow me to add so many lines that aren't in commit.

File: generators/src/main/java/peoplegenerator/CookieEmailGenerator.java [hunk 5]
```diff
@@ -11,33 +11,27 @@
...
             }
-        } catch (IOException e){
+        } catch (IOException e) {
             e.printStackTrace();
         }
     }
 
     /**
      * Example: 1000 generators/src/main/resources/people.csv generators/src/main/resources/userCookieEmailMappings.csv
      */
```
Comment (S4RPYK3IVl5) [inline generators/src/main/java/peoplegenerator/CookieEmailGenerator.java:33]: Comment

File: generators/src/main/java/peoplegenerator/CookieEmailGenerator.java [hunk 6]
```diff
@@ -11,33 +11,27 @@
...
 
-        while ((nextLine = reader.readNext()) != null) {
-            list.add(nextLine[4]);
-        }
-
-        try(FileWriter writer = new FileWriter(pathToCookieEmail, false)) {
+        try (FileWriter writer = new FileWriter(pathToCookieEmail, false)) {
             for (int i = 0; i <= n; i++) {
-                String data = UUID.randomUUID().toString() + "," + list.get(random.nextInt(list.size())) + ","
+                String data = UUID.randomUUID().toString() + "," + emails.get(random.nextInt(emails.size())) + ","
```
Comment (S4RPYK3IVl5) [inline generators/src/main/java/peoplegenerator/CookieEmailGenerator.java:21]: Use StringBuilder here

**Review by S4RPYK3IVl5 — CHANGES_REQUESTED** [review]

File: generators/src/main/java/kafka/pageclickgenerator/Main.java (hunk 4 above)
Comment (Levshunovma) [reply generators/src/main/java/kafka/pageclickgenerator/Main.java:15]: Slava said that we'll refactor it later

**Review by Levshunovma — APPROVED** [review]
We'll do some minor fixes later

**Review by S4RPYK3IVl5 — APPROVED** [review]

**Pull request merged**